### PR TITLE
Map all mediasoup objects using observer and fix prometheus

### PR DESCRIPTION
**Pull request by Astagor**
I hope this fixes prometheus. Now we have all mediasuoup mappings within appData maps

### CODE REVIEW

File: server/lib/Room.js [hunk 1]
```diff
@@ -1124,37 +1107,21 @@ class Room extends EventEmitter
 				}
 				catch (error)
 				{
-					logger.error(
+					throw new Error(
```
Comment (Astagor) [inline server/lib/Room.js:1110]: @so010 have a look at this change, I think we had an error here as when the creation of the producer failed we have only logged it and the code below was working as if the producer was created. I have changed it to throw an error now

Comment (sd4v1d) [reply server/lib/Room.js:1110]: I tested it and I can confirm that  "not iterable" error is gone, however it seems that consumer/producer metrics are not collected/exported. For some strange reason it is like "workers" map (in lib/stats/metrics/default.js) is sometimes empty.

Comment (Astagor) [reply server/lib/Room.js:1110]: @sd4v1d I had a look at the code, and I would say it cannot be what workers is empty. How can I configure and run prometheus to reproduce?

Comment (sd4v1d) [reply server/lib/Room.js:1110]: It is working after last commit.

**Review by sd4v1d — APPROVED** [review]
Tested, and it is working as expected. Thanks @Astagor 
@so010 could you please also review and approve PR?

File: server/lib/Room.js (hunk 1 above)
Comment (so010) [reply server/lib/Room.js:1110]: I think that's right if we throw a error here. It is caught on higher level and logged there

**Review by so010 — APPROVED** [review]
:)

**Pull request merged**